lsp-lens.el

```diff
@@ -309,12 +309,19 @@ CALLBACK - callback for the lenses."
           (funcall callback lsp-lens--backend-cache lsp--cur-version)
         (lsp-lens--backend-fetch-missing lsp-lens--backend-cache callback lsp--cur-version)))))
 
+(defun lsp-lens--refresh-buffer ()
+  "Trigger lens refresh on buffer."
+  (remove-hook 'lsp-on-idle-hook #'lsp-lens--refresh-buffer t)
+  (lsp-lens-refresh t (current-buffer)))
+
 (defun lsp--lens-on-refresh (workspace)
   "Clear lens within all buffers of WORKSPACE, refreshing all workspace buffers."
   (cl-assert (not (eq nil workspace)))
-  (cl-loop
-   for ws-buffer in (lsp--workspace-buffers workspace) do
-   (lsp-lens-refresh t ws-buffer)))
+  (->> (lsp--workspace-buffers workspace)
+       (mapc (lambda (buffer)
+               (lsp-with-current-buffer buffer
+                 (add-hook 'lsp-on-idle-hook #'lsp-lens--refresh-buffer nil t)
```

Comment:
I think that here you should call refresh for the visible buffer or in case it is not visible, you should attach the hook as you do. Also, I suspect that you have to check if lens mode is enabled in the buffers that you refresh the lenses.

Reply:
Done!